interfaces/builtin/spi_test.go

```diff
@@ -0,0 +1,186 @@
+    path: /dev/spi-foo
+  bad-spi-6:
+    interface: spi
+`, nil)
+	s.slotGadget1 = &interfaces.Slot{SlotInfo: info.Slots["spi-1"]}
+	s.slotGadget2 = &interfaces.Slot{SlotInfo: info.Slots["spi-2"]}
+	s.slotGadgetBad1 = &interfaces.Slot{SlotInfo: info.Slots["bad-spi-1"]}
+	s.slotGadgetBad2 = &interfaces.Slot{SlotInfo: info.Slots["bad-spi-2"]}
+	s.slotGadgetBad3 = &interfaces.Slot{SlotInfo: info.Slots["bad-spi-3"]}
+	s.slotGadgetBad4 = &interfaces.Slot{SlotInfo: info.Slots["bad-spi-4"]}
+	s.slotGadgetBad5 = &interfaces.Slot{SlotInfo: info.Slots["bad-spi-5"]}
+	s.slotGadgetBad6 = &interfaces.Slot{SlotInfo: info.Slots["bad-spi-6"]}
+
+	info = snaptest.MockInfo(c, `
+name: consumer
+plugs:
+  spi-1:
+    interface: spi
+    path: /dev/spidev.0.0
+  spi-2:
+    interface: spi
+    path: /dev/spidev0.1
+apps:
+  app:
+    command: foo
+    plugs: [spi-1]
+`, nil)
+	s.plug1 = &interfaces.Plug{PlugInfo: info.Plugs["spi-1"]}
+	s.plug2 = &interfaces.Plug{PlugInfo: info.Plugs["spi-2"]}
+}
+
+func (s *spiInterfaceSuite) TestName(c *C) {
+	c.Assert(s.iface.Name(), Equals, "spi")
+}
+
+func (s *spiInterfaceSuite) TestSanitizeSlot(c *C) {
+	c.Assert(s.slotOs1.Sanitize(s.iface), IsNil)
+	c.Assert(s.slotOs2.Sanitize(s.iface), IsNil)
+	c.Assert(s.slotGadget1.Sanitize(s.iface), IsNil)
+	c.Assert(s.slotGadget2.Sanitize(s.iface), IsNil)
+	err := s.slotGadgetBad1.Sanitize(s.iface)
+	c.Assert(err, ErrorMatches, `"/dev/spev0.0" is not a valid SPI device`)
+	err = s.slotGadgetBad2.Sanitize(s.iface)
+	c.Assert(err, ErrorMatches, `"/dev/sidv0.0" is not a valid SPI device`)
+	err = s.slotGadgetBad3.Sanitize(s.iface)
+	c.Assert(err, ErrorMatches, `"/dev/slpiv0.3" is not a valid SPI device`)
+	err = s.slotGadgetBad4.Sanitize(s.iface)
+	c.Assert(err, ErrorMatches, `"/dev/sdev-00" is not a valid SPI device`)
+	err = s.slotGadgetBad5.Sanitize(s.iface)
+	c.Assert(err, ErrorMatches, `"/dev/spi-foo" is not a valid SPI device`)
+	err = s.slotGadgetBad6.Sanitize(s.iface)
+	c.Assert(err, ErrorMatches, `slot "gadget:bad-spi-6" must have a path attribute`)
+	slot := &interfaces.Slot{SlotInfo: &snap.SlotInfo{
+		Snap:      &snap.Info{SuggestedName: "some-snap"},
+		Name:      "spi",
+		Interface: "spi",
+	}}
+	c.Assert(slot.Sanitize(s.iface), ErrorMatches,
+		"spi slots are reserved for the core and gadget snaps")
+}
+
+func (s *spiInterfaceSuite) TestUDevSpec(c *C) {
+	spec := &udev.Specification{}
+	c.Assert(spec.AddConnectedPlug(s.iface, s.plug1, nil, s.slotGadget1, nil), IsNil)
+	c.Assert(spec.Snippets(), HasLen, 1)
+	c.Assert(spec.Snippets()[0], Equals, `KERNEL=="spidev0.0", TAG+="snap_consumer_app"`)
+}
+
+func (s *spiInterfaceSuite) TestAppArmorSpec(c *C) {
+	spec := &apparmor.Specification{}
+	c.Assert(spec.AddConnectedPlug(s.iface, s.plug1, nil, s.slotGadget1, nil), IsNil)
+	c.Assert(spec.SecurityTags(), DeepEquals, []string{"snap.consumer.app"})
+	c.Assert(spec.SnippetForTag("snap.consumer.app"), Equals, "/dev/spidev0.0 rw,/sys/devices/platform/**/**.spi/**/spidev0.0/** rw,")
```

Comment:
You are missing a newline here. This test is saying that you expect:
```
/dev/spidev0.0 rw,/sys/devices/platform/**/**.spi/**/spidev0.0/** rw,
```

but you get:
```
/dev/spidev0.0 rw,
/sys/devices/platform/**/**.spi/**/spidev0.0/** rw,
```

Change it too:
```
c.Assert(spec.SnippetForTag("snap.consumer.app"), Equals, "/dev/spidev0.0 rw\n,/sys/devices/platform/**/**.spi/**/spidev0.0/** rw,")
```

and this test should pass.